### ISSUE TITLE
分类示例在 CrossEntropyLoss 前重复应用了 Softmax / Softmax applied before CrossEntropyLoss

### ISSUE THREAD

**Issue by Mayukhga83** [lansinuote/Huggingface_Toturials#6]
在以下分类示例中：

7.中文分类.ipynb
9.中文句子关系推断.ipynb
11.中文分类_CUDA.ipynb

Model.forward() 中先对分类器输出执行了：

out = out.softmax(dim=1)

但训练阶段随后使用：

criterion = torch.nn.CrossEntropyLoss()
loss = criterion(out, labels)

torch.nn.CrossEntropyLoss 期望输入为未归一化的 logits，并且内部已经执行与 LogSoftmax + NLLLoss 等价的计算。

因此目前的计算实际相当于：

CrossEntropyLoss(softmax(logits), labels)

这会改变损失函数以及训练时的梯度。

建议让 Model.forward() 直接返回 logits：

out = self.fc(out.last_hidden_state[:, 0])
return out

如果推理阶段需要类别概率，可以在训练损失计算之外使用：

probs = torch.softmax(logits, dim=1)

对于当前代码中的 argmax 分类结果，移除 softmax 不会改变预测类别，因为 softmax 保持 logits 的排序。

如果这是有意的实现方式，也许可以在教程中补充说明；否则建议移除模型 forward() 中的 softmax。
